client/landing/gutenboarding/path.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * External dependencies
+ */
+import { generatePath } from 'react-router-dom';
+import { getLanguageSlugs } from '../../lib/i18n-utils';
+import { ValuesType } from 'utility-types';
+
+export const Step = {
+	IntentGathering: 'about',
+	DesignSelection: 'design',
+	PageSelection: 'pages',
+	Signup: 'signup',
+	CreateSite: 'create-site',
+} as const;
+
+const langs: string[] = getLanguageSlugs();
+const steps = Object.keys( Step ).map( key => Step[ key as keyof typeof Step ] );
+
+export const path = `/:step(${ steps.join( '|' ) })?/:lang(${ langs.join( '|' ) })?`;
```

Comment:
This deserves some deliberation IMO. I'm not a big fan of language-last routing (`/some/route/to/something/de`) as we have it in the current signup flow. I think people tend to add more specialized route fragments to the end, which can lead to messy routes (`/some/route/to/something/de/ohwait/ihavesomethingelse`); especially when not everyone is aware of fragment 'whitelists'/'templates' (not sure what they're called) like our `getLanguageSlugs().join( '|' )` here, and how to use that in combination with conditional route fragments (`?`).

My vote for keeping the language fragment first. I think that's also more consistent with industry standards. It should work fine as a conditional fragment since we are using the 'filter'.

Reply:
We talked via other channels and this isn't possible because of the way sections work.

`/SECTION/rest/of/path`

We concluded that "language 2nd" isn't as appealing and "language last" seems fine.